src/main/java/com/ioet/bpm/skills/boundaries/SkillController.java

```diff
@@ -1,65 +1,105 @@
+        List<Skill> skillWithNameCoincidences = skillRepository.findByNameContaining(skillName);
+        return new ResponseEntity<>(skillWithNameCoincidences, HttpStatus.OK);
+    }
+
+    @ApiOperation(value = "Create a new skill", response = Skill.class)
+    @ApiResponses(value = {
+            @ApiResponse(code = 201, message = "Skill successfully created")
+    })
+    @PostMapping(produces = "application/json")
     public ResponseEntity<Skill> createSkill(@RequestBody Skill skill) {
         Skill skillCreated = skillRepository.save(skill);
         return new ResponseEntity<>(skillCreated, HttpStatus.CREATED);
     }
 
-    @DeleteMapping("/{id}")
+    @ApiOperation(value = "Delete a skill")
+    @ApiResponses(value = {
+            @ApiResponse(code = 204, message = "Skill successfully deleted"),
+            @ApiResponse(code = 404, message = "The skill to delete was not found")
+    })
+
+    @DeleteMapping(path = "/{id}", produces = "application/json")
     public ResponseEntity<?> deleteSkill(@PathVariable(value = "id") String skillId) {
         Optional<Skill> skill = skillRepository.findById(skillId);
-        if (!skill.isPresent()) {
-            return new ResponseEntity<>(HttpStatus.NOT_FOUND);
+        if (skill.isPresent()) {
+            skillRepository.delete(skill.get());
+            return new ResponseEntity<>(HttpStatus.OK);
         }
-
-        skillRepository.delete(skill.get());
-        return new ResponseEntity<>(HttpStatus.OK);
+        return new ResponseEntity<>(HttpStatus.NOT_FOUND);
     }
 
-    @PutMapping("/{id}")
+    @ApiOperation(value = "Update a skill", response = Skill.class)
+    @ApiResponses(value = {
+            @ApiResponse(code = 200, message = "Skill successfully updated"),
+            @ApiResponse(code = 404, message = "The skill to update was not found")
+    })
+    @PutMapping(path = "/{id}", produces = "application/json")
     public ResponseEntity<?> updateSkill(@PathVariable(value = "id") String skillId,
                                              @Valid @RequestBody Skill skillDetails) {
-        Optional<Skill> skillFromDB = skillRepository.findById(skillId);
-        if (skillFromDB.isPresent()) {
-            skillDetails.setId(skillFromDB.get().getId());
-            Skill skillSaved = skillRepository.save(skillDetails);
-            return new ResponseEntity<>(skillSaved, HttpStatus.OK);
+
+        Optional<Skill> skillOptional = skillRepository.findById(skillId);
+        if (!skillOptional.isPresent()) {
+            return new ResponseEntity<>(HttpStatus.NOT_FOUND);
         }
 
-        return new ResponseEntity<>("Skill with id " + skillId + " does not exist.", HttpStatus.NOT_FOUND);
+        Skill skill = skillOptional.get();
+        skill.setName(skillDetails.getName());
+        skill.setBusinessValue(skillDetails.getBusinessValue());
+        skill.setPredictiveValue(skillDetails.getPredictiveValue());
+        skillRepository.save(skill);
```

Comment:
You can do instead:

skillDetails.setId(skill.getId());

and then save skillDetails.